core/domain/exp_services_test.py

```diff
@@ -7625,6 +7627,395 @@ def test_revert_exploration_with_mismatch_of_versions_raises_error(
+                                'content_id': 'ca_choices_7',
+                                'html': '<p>3</p>'
+                            },
+                            {
+                                'content_id': 'ca_choices_8',
+                                'html': '<p>4</p>'
+                            }
+                        ]
+                    },
+                    'showChoicesInShuffledOrder': {
+                        'value': False
+                    },
+                    'maxAllowableSelectionCount': {
+                        'value': 1
+                    },
+                    'minAllowableSelectionCount': {
+                        'value': 1
+                    }
+                }
+            }),
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'state_name': 'Introduction',
+                'property_name': 'answer_groups',
+                'new_value': [{
+                    'outcome': { 
+                        'dest': 'End',
+                        'dest_if_really_stuck': None,
+                        'param_changes': [],
+                        'refresher_exploration_id': None,
+                        'missing_prerequisite_skill_id': None,
+                        'feedback': {
+                            'html': '<p>correct</p>',
+                            'content_id': 'feedback_9'},
+                        'labelled_as_correct': True
+                    },
+                    'rule_specs': [{
+                        'rule_type': 'Equals',
+                        'inputs': {
+                            'x': 0
+                        }
+                    }],
+                    'training_data': [],
+                    'tagged_skill_misconception_id': None
+                }],
+            }),
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'property_name': exp_domain.STATE_PROPERTY_INTERACTION_HINTS,
+                'state_name': exploration.init_state_name,
+                'new_value': [
+                    {
+                        'hint_content': {
+                            'content_id': 'hint_10',
+                            'html': '<p>Hint 1</p>'
+                        }
+                    },
+                    {
+                        'hint_content': {
+                            'content_id': 'hint_11',
+                            'html': '<p>Hint 2</p>'
+                        }
+                    }
+                ]
+            }),
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                'new_value': 12,
+                'old_value': 11,
+                'property_name': 'next_content_id_index'
+            })]
+
+        exp_services.update_exploration(
+            self.owner_id, self.EXP_0_ID, change_list, 'Update 1')
```

Comment:
Shouldn't we verify translation counts after this step?

Reply:
done